org.eclipse.jdt.ui.tests.refactoring/resources/ExtractTemp/canExtract/A_test110_out.java

```diff
@@ -2,11 +2,12 @@
 
 class A {
 	void m() {
+		System.out.println(calculateCount());
```

Comment:
OK. I updated the test case, and the difference from before the PR is that if the selected expression is the only direct child of the statement, only the selected expression would be extracted.

Reply:
Ok, I figured out why my results and yours are different.  I tried converting the first instance of calculateCount() and I get the old result.  The test is actually converting the 2nd calculateCount() and when I do this, I get your version.  I don't believe the behavior should differ in this case because I asked for all instances and the method call in question doesn't change any state so the result should have put the temp in front of the first call as requested.

Reply:
Hello, I restored the processing logic to the original version. However, here is a bug in the original version when it comes to such an example:
```java
1 {
2   f();
3   int x=f();
4 }
5 {
6   int y=f();
7 }
```
If we select `f()` at Line 2, the refactored code shown as follows has obviously syntax errors: 
```java
1 {
2   int f=f();
3   int x=f;
4 }
5 {
6   int y=f;
7 }
```
I suspect there is a problem with the validation of the first occurrence of the expression. Do you think this problem needs to be fixed? One solution is to compare the lowest common ancestor node from all extracted nodes and the scope of the new variable, but it will increase the extra computational cost.

Reply:
Yes, it needs to be fixed.  Your example above isn't common, however, if you change it to:

```
if (x < 3) {
   f();
   int x = f();
} else {
   int y = f();
}
```

you'll see the same error caused.  I am fine with the code simply not changing the int y = f() statement (as opposed to moving the f = f(); statement before the first block) as that should be simpler to recognize.

BTW: everything else looks very good.

Reply:
Hi, I fixed the bug following your suggestion. Test-149 and Test-150 are appended to validate the correctness of the patch.

Reply:
Ok thx.